src/core/openthread-core-default-config.h

```diff
@@ -1464,4 +1464,45 @@
 #define OPENTHREAD_CONFIG_DISABLE_CCA_ON_LAST_ATTEMPT 0
 #endif
 
+/**
+ * @def OPENTHREAD_CONFIG_ENABLE_TIME_SYNC
+ *
+ * Define as 1 to enable the time synchronization service feature.
+ *
+ */
+#ifndef OPENTHREAD_CONFIG_ENABLE_TIME_SYNC
+#define OPENTHREAD_CONFIG_ENABLE_TIME_SYNC 0
+#endif
+
+/**
+ * @def OPENTHREAD_CONFIG_TIME_SYNC_ESSENTIAL
+ *
+ * Define as 1 to make the time synchronization feature essential, then the device will form a new
+ * time synchronization enabled network when there is no existing time sync enabled network around.
```

Comment:
suggestion: "Define as 1 tor require time synchronization when attaching to a network.  If the device is router capable and cannot find a neighboring router supporting time synchronization, the device will form a new partition.  If the device is not router capable, the device will remain an orphan."